### PR TITLE
AutoDJ: make 'enable' shortcut work after startup 

### PR DESCRIPTION
**Pull request by ronso0**
I noticed two issue I need help with.
and I rather discuss them here to work around launchpad and ease code references.

**1)** before the AutoDJ hotkey Shift+F12 would toggle [AutoDJ],enable after startup any library feature must have had focus (Ctrl+F to focus the searchbar is sufficient, just ANY trackmodel needs to be selected?)

**2)** with 'Allow random track addition' disabled the above wouldn't work either:
- add some tracks to the AutoDJ queue, close Mixxx
- start Mixxx
- Ctrl+F to focus the searchbar
- press Shift + F12 to start AutoDJ
= log "Debug[Main] Queue is empty now"
from https://github.com/mixxxdj/mixxx/blob/2.3/src/library/autodj/autodjprocessor.cpp#L403
because getNextTrackFromQueue() returns 0 even though the queue holds enough tracks.
- click AutoDJ feature
- press Shift + F12 to start AutoDJ
= AutoDJ starts

**2** is solved with change added here: m_pAutoDJTableModel->select(); in autodjprocessor.
Though this is just a hack, I have no idea about the implications.

For **1** I have no idea..

### CODE REVIEW

**Comment by daschuer** [comment]
A controller that has mapped [AutoDJ],enabled is also affected. 

I think that it is sufficient that m_pAutoDJTableModel->select(); is called once.

**Comment by ronso0** [comment]
what about

> 1) before the AutoDJ hotkey Shift+F12 would toggle [AutoDJ],enable after startup any library feature must have had focus

I doubt the change here fixes it for controllers as the shortcut is not working.
Where else do we need to m_pAutoDJTableModel->select(); ?

File: src/library/autodj/autodjprocessor.cpp [hunk 1]
```diff
@@ -829,6 +829,7 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
 
 TrackPointer AutoDJProcessor::getNextTrackFromQueue() {
     // Get the track at the top of the playlist.
+    m_pAutoDJTableModel->select();
```
Comment (daschuer) [inline src/library/autodj/autodjprocessor.cpp:832]: The Shift+F12 issue after start-up is also solved if this is moved here: 
https://github.com/mixxxdj/mixxx/blob/1ca77e00a53bbd63e5bb899ac27b8897efa95200/src/library/autodj/autodjprocessor.cpp#L126

Comment (daschuer) [reply src/library/autodj/autodjprocessor.cpp:832]: And you need to add 
```
installEventFilter(m_pKeyboard);
```
here: 
https://github.com/mixxxdj/mixxx/blob/a74230c8c478e72e4d1af7ad0a3c0a2eb09388fe/src/mixxx.cpp#L164

Comment (ronso0) [reply src/library/autodj/autodjprocessor.cpp:832]: > ```
> mixxx.cpp#L164: installEventFilter(m_pKeyboard);
> ```

that did the trick! Thanks

Comment (ronso0) [reply src/library/autodj/autodjprocessor.cpp:832]: even though it's there, too https://github.com/mixxxdj/mixxx/blob/2.3/src/mixxx.cpp#L538

**Comment by ronso0** [comment]
@daschuer Works beautifully, ready for review!

**Review by daschuer — APPROVED** [review]
LGTM, Thank you for taking care.

**Pull request merged**